numpy/core/src/multiarray/scalarapi.c

```diff
@@ -390,42 +391,51 @@ PyArray_ScalarFromObject(PyObject *object)
             PyArrayScalar_RETURN_FALSE;
         }
     }
-    else if (PyInt_Check(object)) {
-        ret = PyArrayScalar_New(Long);
-        if (ret == NULL) {
-            return NULL;
+    else if (PyLong_Check(object)) {
+        npy_longlong val;
+
+        /* Check if fits in long */
+        val = (npy_longlong)PyLong_AsLong(object);
+        if (!error_converting(val)) {
+            ret = PyArrayScalar_New(Long);
+            if (ret != NULL) {
+                PyArrayScalar_VAL(ret, Long) = (npy_long)val;
```

Comment:
This casting feels a little silly, perhaps better to use a separate `npy_long` variable.

Reply:
Can do.

Reply:
Done.